### ISSUE TITLE
Date parsing issue

### ISSUE THREAD

**Issue by fontikar** [traitecoevo/infinitylists#11]
Data from ALA currently consists of mixture of dmy and mdy dates. 
<img width="780" alt="Screenshot 2023-08-15 at 9 45 58 am" src="https://github.com/traitecoevo/plant-map-app/assets/12161802/0351874f-657d-49a2-82c2-3a0b276826ff">

This effects the "pulling more recent record" function

**Comment by wcornwell** [comment]
the issue seems to be coming from here: https://github.com/traitecoevo/plant-map-app/blob/11484c06a41ea8219eb8c973cca159b803fbdd9e/ala_processing.R#L36. might be related to time zones.

**Comment by wcornwell** [comment]
there are two outstanding issues with respect to dates:

1) we're not handling timezones very well, which is making things off by one day in some cases
2) the date sorting thing in datatable doesn't work with this date format.

**Comment by wcornwell** [comment]
```
NSW_plants_tzdate  <- NSW_plants_renamed |> 
  mutate(collectionDate = ymd_hms(eventDate) |> with_tz())
```
seems to work well, but we're losing some data at this step

**Comment by wcornwell** [comment]
<img width="380" alt="Screenshot 2023-08-22 at 5 13 39 pm" src="https://github.com/traitecoevo/infinitylists/assets/5534161/162f1ee6-d917-42c4-9694-d7a74d4fb0d5">

**Comment by wcornwell** [comment]
some parsing failures, that lead to data dropped later on.  

<img width="384" alt="Screenshot 2023-08-22 at 5 14 28 pm" src="https://github.com/traitecoevo/infinitylists/assets/5534161/f98e8267-1293-48bf-aeef-27b1bf8cdecc">

maybe that's OK, not sure

**Comment by wcornwell** [comment]
some of them are like this to start with "2016-04-06 UTC" and others are like this ""2021-12-22 09:48:00 UTC"

**Comment by fontikar** [comment]
Hmm is this for plant data or tassie butterflies

**Comment by wcornwell** [comment]
Butterflies.  but I think I fixed it on the generalization branch

**Comment by fontikar** [comment]
I think some dates dates do get parsed oddly by ALA so sometimes its good to grab verbatimeventDate just in case

**Comment by fontikar** [comment]
So in the tibble, you can't see the timezone info. I had to use $ to see

![Screenshot 2023-08-22 at 5 43 42 pm](https://github.com/traitecoevo/infinitylists/assets/12161802/9d29d22f-03da-4f7a-92a1-f765ea4a8598)

**Comment by fontikar** [comment]
> some parsing failures, that lead to data dropped later on.
> 
> <img alt="Screenshot 2023-08-22 at 5 14 28 pm" width="384" src="https://user-images.githubusercontent.com/5534161/262265324-f98e8267-1293-48bf-aeef-27b1bf8cdecc.png">
> maybe that's OK, not sure

I think the 6k records that get dropped may not have UTC in the eventdate? 
Need to go back and check

**Comment by wcornwell** [comment]
i think this is working as far as i can tell

**Comment by fontikar** [comment]
Reopenning this so I can try fix it again!

**Comment by fontikar** [comment]
Addressed in [fe57d18](https://github.com/traitecoevo/infinitylists/commit/fe57d18c498c494b853b4bd977de491d8cd436e2)
